### PR TITLE
tools: data verification tool, second ten files

### PR DESCRIPTION
**Pull request by ooooooops**
second ten files

### CODE REVIEW

File: modules/map/tools/map_datachecker/server/pose_collection_agent.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  void on_bestgnsspos_callback(
+      const std::shared_ptr<const apollo::drivers::gnss::GnssBestPose>
+          &bestgnsspos);
+  std::shared_ptr<std::vector<FramePose>> get_poses() const;
+
+ private:
+  void reset();
+
+ private:
+  std::mutex _mutex;
```
Comment (deidaraho) [inline modules/map/tools/map_datachecker/server/pose_collection_agent.h]: member variable name: mutex_

File: modules/map/tools/map_datachecker/server/common.h [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  /*loops to check*/
+  size_t laps_number;
+  /*addtional loops to check*/
+  int laps_number_additional;
+  /*Proportional thresholds for all required points
+  in acquisition cycle checking*/
+  double laps_rate_thresh;
+};
+
+inline std::shared_ptr<JSonConf> parse_json(std::string conf_path) {
```
Comment (xiaoxq) [inline modules/map/tools/map_datachecker/server/common.h]: Remove `inline` for non-trivial functions.

File: modules/map/tools/map_datachecker/server/loops_verify_agent.cc [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+  }
+}
+
+std::shared_ptr<std::vector<std::pair<double, double>>>
+LoopsVerifyAgent::get_verify_range(LoopsVerifyRequest *request) {
+  std::shared_ptr<std::vector<std::pair<double, double>>> sp_range(
+      new std::vector<std::pair<double, double>>());
+  int range_size = request->range_size();
+  for (int i = 0; i < range_size; ++i) {
+    const ::apollo::hdmap::VerifyRange &range = request->range(i);
```
Comment (xiaoxq) [inline modules/map/tools/map_datachecker/server/loops_verify_agent.cc]: Try `for (const VerifyRange& range : request->range())`, which is shorter, more efficient and self declaration.

File: modules/map/tools/map_datachecker/server/loops_verify_agent.cc [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+  }
+
+  double progress = sp_laps_checker_->get_progress();
+  response->set_progress(progress);
+  response->set_code(ErrorCode::SUCCESS);
+  if (std::abs(1.0 - progress) < 1e-8) {
+    double confidence = sp_laps_checker_->get_confidence();
+    size_t lap = sp_laps_checker_->get_lap();
+    AINFO << "aquired lap: " << lap << ", conf: " << confidence;
+    apollo::hdmap::LoopResult *loop_result = response->mutable_loop_result();
```
Comment (xiaoxq) [inline modules/map/tools/map_datachecker/server/loops_verify_agent.cc]: You don't need the `apollo::hdmap::` prefix as it's inside the namespace.

**Comment by xiaoxq** [comment]
Close as the new pull request has been merged.

**Pull request closed without merging**